hbase_sql/src/main/java/com/splicemachine/olap/OlapServerSubmitter.java

```diff
@@ -146,18 +149,22 @@ public void run() {
             } else {
                 this.appStagingBaseDir = FileSystem.get(conf).getHomeDirectory();
             }
-            String yarnQueue = System.getProperty("splice.olapServer.yarn.queue");
+            String yarnQueue = conf.get("splice.olapServer.yarn.queue");
             if (yarnQueue == null) {
-                yarnQueue = System.getProperty("splice.spark.yarn.queue", YARN_DEFAULT_QUEUE_NAME);
+                yarnQueue = conf.get("splice.spark.yarn.queue", YARN_DEFAULT_QUEUE_NAME);
             }
-
+            YarnApplicationState appState = YarnApplicationState.NEW;
             for (int i = 0; i<maxAttempts; ++i) {
                 try {
                     ApplicationId appId = findApplication(yarnClient);
-                    if (appId == null) {
+                    if (appId == null || (rollingUpgrade && appState != YarnApplicationState.FINISHED)) {
+                        SpliceLogUtils.info(LOG, "Submitting a new OLAP server for rolling upgrade");
                         appId = submitApplication(yarnClient, memory, memoryOverhead, olapPort, cpuCores, sparkYarnQueue, yarnQueue);
                     }
-
+                    else {
+                        // During rolling upgrade, another submitter launched an olap server
```

Comment:
This breaks existing logic, if an HMaster restarts (or fails over) we won't keep pinging the still running OlapServer so it'll eventually exit too.